### PR TITLE
initial Playground SDK v0.1.0

### PR DESCRIPTION
**Pull request by xke**

### CODE REVIEW

File: packages/playground/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    };
+    this.registeredCustomObjects.push(customObject);
+    /*
+    console.log(
+      "registeredCustomObjects: " + JSON.stringify(this.registeredCustomObjects)
+    );
+    */
+  }
+
+  public registerInteraction(
```
Comment (viveknair) [inline packages/playground/src/index.ts:213]:
```
public registerInteraction<FieldType>(
  name: string,
  inputFields: FieldType,
  outputFields: object,
  interaction: InteractionFunc<{ [K in keyof FieldType]: any }>,
) {
  ... 
}
```

Can we do more clever typing here to improve DX similar to the above (not sure if that actually works)? No worries for the first version, but definitely for GA, we'll need to refine these types to be more user friendly.

At the very least, we shouldn't `any` type the interaction function.

File: packages/playground/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
+import { getGentraceApiKey } from "@gentrace/core";
+import { WebSocket } from "ws";
+import { v4 as uuidv4 } from "uuid";
+
+export { init } from "@gentrace/core"; // for accessing the Gentrace API key
+
+import { AsyncLocalStorage } from "async_hooks";
+const asyncLocalStorage = new AsyncLocalStorage();
+
+interface customObject {
```
Comment (viveknair) [inline packages/playground/src/index.ts:10]: Let's use types not interfaces whenever possible. We want to be explicit about what key/value pairs are required. Typescript interfaces are much too permissive.

File: packages/playground/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      init: "pg-sdk",
+      data: {
+        type: "setup",
+        apiKey: getGentraceApiKey(),
+        interactions: this.registeredInteractionObjects,
+        types: this.formatSetupTypes(this.registeredCustomObjects),
+      },
+    };
+
+    const ws = new WebSocket(this.WEBSOCKET_URL);
```
Comment (viveknair) [inline packages/playground/src/index.ts:98]: I don't know if @dsafreno implemented a ping event on the server-side, but I've run into the issue documented by the below link a few times before when working with WebSockets. This isn't a blocker but probably appropriate for GA.

https://github.com/websockets/ws#how-to-detect-and-close-broken-connections

File: packages/playground/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+import { AsyncLocalStorage } from "async_hooks";
+const asyncLocalStorage = new AsyncLocalStorage();
+
+interface customObject {
+  typeName: string;
+  objectName: string;
+  object: object;
+}
+
+interface interactionObject {
```
Comment (viveknair) [inline packages/playground/src/index.ts:16]: General convention with type or interface definitions in TypeScript is to use [PascalCase](https://www.theserverside.com/definition/Pascal-case)

**Review by viveknair — CHANGES_REQUESTED** [review]

File: packages/playground/src/index.ts (hunk 4 above)
Comment (xke) [reply packages/playground/src/index.ts]: [commit](https://github.com/gentrace/gentrace-node/pull/337/commits/4581b6aba21976ea82b4ab9415f9891a58c235c9)

File: packages/playground/src/index.ts (hunk 2 above)
Comment (xke) [reply packages/playground/src/index.ts]: [commit](https://github.com/gentrace/gentrace-node/commit/53b1aa986c43fd3d1aa18c0cad09af13112732cd)

**Review by viveknair — APPROVED** [review]

File: packages/playground/src/index.ts (hunk 3 above)
Comment (xke) [reply packages/playground/src/index.ts:94]: [this commit](https://github.com/gentrace/gentrace-node/pull/341/commits/265178c9ba8b1b2fa8296f3e23f1a90a72b6394e) has a ping version that @dsafreno had used. I tested it and it seems to keep the connection from breaking too quickly on Staging.

File: packages/playground/src/index.ts (hunk 1 above)
Comment (xke) [reply packages/playground/src/index.ts:203]: https://github.com/gentrace/gentrace-node/pull/384

**Pull request merged**